### PR TITLE
Show app review dialog after completing co insured self serve

### PR DESCRIPTION
**Pull request by hugokallstrom**

### CODE REVIEW

File: app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/data/CommitMidtermChangeUseCase.kt [hunk 1]
```diff
@@ -35,6 +37,7 @@ internal class CommitMidtermChangeUseCaseImpl(
     when (val state = result.midtermChangeIntentCommit.intent?.state) {
       MidtermChangeIntentState.COMPLETED -> {
         networkCacheManager.clearCache()
+        selfServiceCompletedEventStore.onSelfServiceCompleted()
```
Comment (hugokallstrom) [inline app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/data/CommitMidtermChangeUseCase.kt:40]: Will add this to moving flow and claim submission as well

File: app/core/core-app-review/src/main/kotlin/com/hedvig/android/core/appreview/ReviewDialogViewModel.kt [hunk 2]
```diff
@@ -12,20 +11,19 @@ import kotlinx.coroutines.flow.receiveAsFlow
...
   val shouldOpenReviewDialog: Flow<Boolean> = _shouldOpenReviewDialog.receiveAsFlow()
 
   init {
     viewModelScope.launch {
-      chatClosedEventStore.observeChatClosedCounter()
-        .map { it % 4 == 0 && it != 0 }
-        .filter { it == true }
+      selfServiceCompletedEventStore.observeNumberOfCompletedSelfServices()
+        .map { it > 0 }
+        .filter { it }
```
Comment (panasetskaya) [inline app/core/core-app-review/src/main/kotlin/com/hedvig/android/core/appreview/ReviewDialogViewModel.kt:23]:
```suggestion
        .filter { it > 0 }
```

Comment (panasetskaya) [reply app/core/core-app-review/src/main/kotlin/com/hedvig/android/core/appreview/ReviewDialogViewModel.kt:23]: Just felt a bit confused :)

**Review by StylianosGakis — APPROVED** [review]
Super good idea.
I took the time to do this https://github.com/HedvigInsurance/android/pull/2041 on top of this just to simplify a bit the fact that we do not need the VM as we had it in the past. Also made it so that it will not try to call the review dialog whenever the app is opened and they may have already done a self service process in the past. This would happen before because it > 0 would be true on startup, and then we'd try to trigger it.

That PR makes it so that whenever a new observer is looking into the flow, the first emission from the Datastore is ignored, as that's the one that just emits whatever there was inside the DataStore itself. And then the *next* emission, which can only happen when a self-service flow is successfully made, then we try to launch the review dialog

**Pull request merged**